src/liseq.py

```diff
@@ -31,4 +31,18 @@ def liseq(x: Sequence[Any]) -> list[int]:
     """
     best: list[int] = []
     # Explore all alternatives
+    for i in range(len(x)):
+        for j in range(i, len(x)):
+            current_list=[i]
+            if is_increasing([x[i],x[j]]):
+                current_list.append(j)
+                j+=1
+                while j<len(x):
+                    if is_increasing([x[j-1],x[j]]):
+                        current_list.append(j)
+                        j+=1
+                    else: 
+                        break
```

Comment:
If you change the condition for the loop outside this, so you check `x[j-1] <= x[j]` there, then you can avoid the inner loop. You just keep increasing `j` as long as it is larger than the previous.